src/azul/plugins/metadata/hca/transform.py

```diff
@@ -840,39 +840,72 @@ def _sequencing_input(self, sequencing_input: api.Biomaterial) -> MutableJSON:
-        sample_ = dict.fromkeys(ChainMap(
-            self._cell_line_types(),
-            self._organoid_types(),
-            self._specimen_types()
-        ).keys())
-        entity_type, entity_dict = (
-            'cell_lines', self._cell_line(sample)
-        ) if isinstance(sample, api.CellLine) else (
-            'organoids', self._organoid(sample)
-        ) if isinstance(sample, api.Organoid) else (
-            'specimens', self._specimen(sample)
-        ) if isinstance(sample, api.SpecimenFromOrganism) else (
-            require(False, sample), None
-        )
-        sample_.update(entity_dict)
-        sample_['entity_type'] = entity_type
-        assert hasattr(sample, 'organ') != hasattr(sample, 'model_organ')
-        sample_['effective_organ'] = sample.organ if hasattr(sample, 'organ') else sample.model_organ
-        assert sample_['document_id'] == str(sample.document_id)
-        assert sample_['biomaterial_id'] == sample.biomaterial_id
-        return sample_
+    def _sample_cell_line(self, cell_line: api.CellLine) -> MutableJSON:
+        return {
+            'organ': None,
+            'organ_part': [],
+            'model_organ': cell_line.model_organ,
+            'model_organ_part': None,
+            'effective_organ': cell_line.model_organ,
+        }
+
+    def _sample_organoid(self, organoid: api.Organoid) -> MutableJSON:
+        return {
+            'organ': None,
+            'organ_part': [],
+            'model_organ': organoid.model_organ,
+            'model_organ_part': organoid.model_organ_part,
+            'effective_organ': organoid.model_organ,
+        }
+
+    def _sample_specimen(self, specimen: api.SpecimenFromOrganism) -> MutableJSON:
+        return {
+            'organ': specimen.organ,
+            'organ_part': sorted(specimen.organ_parts),
+            'model_organ': None,
+            'model_organ_part': None,
+            'effective_organ': specimen.organ,
+        }
+
+    def _samples(self, samples: Iterable[api.Biomaterial]) -> MutableJSON:
+        """
+        Returns inner entity values with all given samples represented both in a
+        'samples' inner entity and a type-specific 'sample_{entity_type}'.
+        The 'samples' inner entity is a minified polymorphic field containing
+        the few properties common to all samples. This allows facets on these
+        properties regardless of the sample entity type.
+        """
+        inner_entities = {'samples': []}
+        entity_types = [
+            ('cell_lines', api.CellLine, self._cell_line, self._sample_cell_line),
+            ('organoids', api.Organoid, self._organoid, self._sample_organoid),
+            ('specimens', api.SpecimenFromOrganism, self._specimen, self._sample_specimen),
+        ]
+        for entity_type, api_class, entity_fn, sample_fn in entity_types:
+            sample_entity_type = f'sample_{entity_type}'
+            inner_entities[sample_entity_type] = []
```

Comment:
Using defaultdict(list) eliminates this

Reply:
The `defaultdict(list)` approach was problematic with the assumption that all sample inner entities (`sample_cell_lines`, `sample_organoids`, `sample_specimens`) would always be in the aggregate when the response was being composed.
This will work however with a slight change to `make_samples()`.